sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/AuthorizationChallengeParser.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.http.policy;
+
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parses Authorization challenges from the {@link HttpResponse}.
+ */
+public class AuthorizationChallengeParser {
+
+    // These patterns are use to parse a challenge string from the WWw-Authenticate header.
+    // A WWW-Authenticate may have more than one challenge, grouped by scheme. The first pattern groups schemes and their parameters.
+    // The second pattern parses the parameters out of a given challenge scheme.
+    private static final Pattern CHALLENGE_PATTERN = Pattern.compile("(\\w+) ((?:\\w+=\"[^\"]*\",?\\s*)+)");
+    private static final Pattern CHALLENGE_PARAMS_PATTERN = Pattern.compile("(\\w+)=\"([^\"]*)\"");
+
+    /**
+     * Examines a {@link HttpResponse} to see if it is a CAE challenge.
+     * @param response The {@link HttpResponse} to examine.
+     * @return True if the response is a CAE challenge, false otherwise.
+     */
+    public static boolean isCaeClaimsChallenge(HttpResponse response) {
+        String error = getChallengeParameterFromResponse(response, "Bearer", "error");
+        return "insufficient_claims".equals(error);
```

Comment:
This is necessary but not sufficient--a CAE challenge also has a "claims" parameter